### DISCUSSION TITLE
Admin Page

### DISCUSSION THREAD

**Discussion by lamba** [markshust/docker-magento#1061] [Q&A]
Hi. I don't know if I missed this in the instructions, but what's the user ID and password for the admin page?

https://magento.test/admin

Update 1

I actually found the information in the magento.env file. But it wanted to do a 2-factor authentication. So, I had to change the email and restart Magento. Still not able to login though.

Update 2

I found that the password is actually stored in the admin_user table. So I edited that and made sure I'm using uppercase, lowercase, numeric, special character, and at least 8 characters. Still no dice! :)

Update 3

I was able to get back to the 2-factor authentication prompt by following this and created a salted password

https://stackoverflow.com/q/72628451/682312

Will Magento be able to email me? I tried to disable to 2fa using the following command, but that didn't work

bin/magento module:disable Magento_TwoFactorAuth

SUCCESS!

I used Mark's module to disable 2fa and was finally able to login to the Admin Console!

https://github.com/markshust/magento2-module-disabletwofactorauth

Still have the q - does 2fa work in the local docker-magento setup? I never did get the email

**Comment by YevhenZvieriev** [comment]
Hi, my friend, I will help you with pleasure.

> but what's the user ID and password for the admin page?

You always can use the ```bin/magento admin:user:create```  to create a new admin user,  or use my [script](https://github.com/markshust/docker-magento/pull/1040), which utilizes the ```bin/magento admin:user:create``` command with predefined values for the admin user.

I created PR, and waiting for my code will be merged.





> I found that the password is actually stored in the admin_user table. So I edited that and made sure I'm using uppercase, lowercase, numeric, special character, and at least 8 characters. Still no dice! :)

Yes, however by merely changing the password in the table, you won't be able to access the admin. The table contains an encrypted password.

To get access to the admin, just disable this ```bin/magento module:disable Magento_TwoFactorAuth```, and don`t forget about dependency with ```Magento_AdminAdobeImsTwoFactorAuth```



> Still have the q - does 2fa work in the local docker-magento setup? I never did get the email

If you mean will you be able to get the letter via personal email, the answer is ***NO!***

But if you pay attention to ```compose.yaml``` there is  ```mailcatcher``` with 1080 port.

So, it intercepts all email letters.

Good luck ^_^
